routers/repo/setting.go

```diff
@@ -313,43 +314,83 @@ func SettingsPost(ctx *context.Context, form auth.RepoSettingForm) {
 
 	case "transfer":
 		if !ctx.Repo.IsOwner() {
-			ctx.Error(404)
+			ctx.Error(http.StatusUnauthorized)
```

Comment:
It should probably be 404 at least for users not having read access to repositiry